pkg/api/types.go

```diff
@@ -123,11 +123,30 @@ type InputConfiguration struct {
 	// build-time dependencies for the project.
 	TestBaseImage *ImageStreamTagReference `json:"test_base_image,omitempty"`
 
+	// BuildRootImage supports two ways to get the image that
+	// the pipeline will caches on. The one way is to take the reference
+	// from an image stream, and the other from a dockerfile.
+	BuildRootImage *BuildRootImageConfiguration `json:"build_root,omitempty"`
+
 	// ReleaseTagConfiguration determines how the
 	// full release is assembled.
 	ReleaseTagConfiguration *ReleaseTagConfiguration `json:"tag_specification,omitempty"`
 }
 
+// BuildRootImageConfiguration holds the two ways of using a base image
+// that the pipeline will caches on.
+type BuildRootImageConfiguration struct {
+	ImageStreamTagReference *ImageStreamTagReference `json:"image_stream_tag,omitempty"`
+	GitSourceReference      *GitSourceReference      `json:"git_source_image,omitempty"`
+}
+
+// GitSourceReference identifies the location of a Dockerfile that will
+// be used as source for a build.
+type GitSourceReference struct {
```

Comment:
This is now even closer to `ProjectDirectoryImageBuildStepConfiguration` -- let's factor out the common fields (`ContextDir`, `DockerfilePath` and `Inputs`) and use struct embedding to share the sub-struct between this use-case and the `ProjectDirectoryImageBuildStepConfiguration`.